frontend/src/forms/manage-restaurant-form/ImageSection.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import { useFormContext } from 'react-hook-form';
+
+const ImageSection = () => {
+  const { control } = useFormContext();
+
+  return (
+    <div className="space-y-2">
+      <div>
+        <h2 className="text-2xl font-bold">Image</h2>
+        <FormDescription>
+          Add a image that will be displayed on your resataurant listing
+        </FormDescription>
+      </div>
+
+      <div className="flex flex-col gap-8 w-[50%]">
+        <FormField
+          control={control}
+          name="imageFile"
+          render={({ field }) => (
+            <FormItem>
+              <FormControl>
+                <Input
+                  className="bg-white"
+                  type="file"
+                  accept=".jpg, .jpeg, .png"
+                  onChange={(event) =>
+                    field.onChange(
+                      event.target.files ? event.target.files[0] : null
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/31#issuecomment-2392842420>#ecff14</a><span> - 10/04/2024, 05:21 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current implementation directly assigns the selected file to the form state, which may cause issues with React's state management and lead to unexpected behavior.
</div>
<div id="fix">
<b>Fix:</b> Modify the onChange handler to create a new object with the selected file, rather than directly assigning it. This ensures proper state updates in React.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -34,7 +34,10 @@
                    onChange={(event) =>
                      field.onChange(
 -                      event.target.files ? event.target.files[0] : null
 +                      event.target.files
 +                        ? { file: event.target.files[0] }
 +                        : null
 +                    )
                    }
                  />
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged